code/modules/cargo/materials_market.dm

```diff
@@ -170,38 +236,53 @@
+				say("Not enough money to start purchase!")
 				return
 
 			var/list/things_to_order = list()
 			things_to_order += (sheet_to_buy)
 			things_to_order[sheet_to_buy] = quantity
+
 			// We want to count how many stacks of all sheets we're ordering to make sure they don't exceed the limit of 10
 			// If we already have a custom order on SSshuttle, we should add the things to order to that order
-			for(var/datum/supply_order/order in SSshuttle.shopping_list)
-				// Must be a Galactic Materials Market order and payed by the null account(if ordered via cargo budget) or by correct user for private purchase
-				if(order.orderer_rank == "Galactic Materials Market" && ( \
-					(!is_ordering_private && order.paying_account == null) || \
-					(is_ordering_private && order.paying_account != null && order.orderer == living_user) \
-				))
-					// Check if this order exceeded its limit
-					var/prior_stacks = 0
-					for(var/obj/item/stack/sheet/sheet as anything in order.pack.contains)
-						prior_stacks += ROUND_UP(order.pack.contains[sheet] / 50)
-						if(prior_stacks >= 10)
-							to_chat(usr, span_notice("You already have 10 stacks of sheets on order! Please wait for them to arrive before ordering more."))
-							playsound(usr, 'sound/machines/synth_no.ogg', 35, FALSE)
-							return
-					// Append to this order
-					order.append_order(things_to_order, cost)
+			var/datum/supply_order/current_order = find_order(living_user, is_ordering_private)
+			if(!isnull(current_order))
+				// Check if this order exceeded the market limit
+				var/prior_sheets = current_order.pack.contains[sheet_to_buy]
+				if(prior_sheets + quantity > SSstock_market.materials_quantity[material_bought] )
+					say("There is not enough sheets in the market! Please wait for them to arrive before ordering more.")
+					playsound(usr, 'sound/machines/synth_no.ogg', 35, FALSE)
+					return
+
+				// Check if the order exceeded the purchase limit
+				var/prior_stacks = ROUND_UP(prior_sheets / MAX_STACK_SIZE)
+				if(prior_stacks >= MAX_STACK_LIMIT)
+					say("There is already 10 stacks of sheets on order! Please wait for them to arrive before ordering more.")
+					playsound(usr, 'sound/machines/synth_no.ogg', 35, FALSE)
+					return
+
+				// Prevents you from ordering more than the available budget
+				var/datum/bank_account/paying_account = account_payable
+				if(!isnull(current_order.paying_account)) //order is already being paid by another account
+					paying_account = current_order.paying_account
+				if(current_order.get_final_cost() + cost > paying_account.account_balance)
+					say("Order exceeds available budget!. Please send it before purchasing more.")
 					return
 
-			//Now we need to add a cargo order for quantity sheets of material_bought.sheet_type
+				// Finally Append to this order
+				current_order.append_order(things_to_order, cost)
+				say("Order appended, total cost is [current_order.get_final_cost()] cr.")
```

Comment:
![image](https://github.com/tgstation/tgstation/assets/41715314/2dc764b5-c142-494c-883b-e2bc58ff3a6e)


This is extremely effective at spamming the chat. I would recommend either using a non-dynamic message, using a dynamic messages and maybe a cooldown for how often they appear, or keeping the dynamic message as is and swapping to to_chat instead.